options/component/app.js

```diff
@@ -0,0 +1,250 @@
+
+import styles from './app.css';
+
+export default class App extends Component {
+
+  state = {
+    augmentationEnabled: true,
+    DAPPS: '',
+    integrationEnabled: true,
+    isPristine: true,
+    lookupURL: '',
+    savedConf: {},
+    showAdvanced: false,
+    UI: ''
+  };
+
+  componentWillMount () {
+    Config.get()
+      .then((config) => {
+        const { augmentationEnabled, DAPPS, integrationEnabled, lookupURL, UI } = config;
+        const conf = { augmentationEnabled, DAPPS, integrationEnabled, lookupURL, UI };
+
+        this.setState({ ...conf, savedConf: conf });
+      });
+  }
+
+  render () {
+    const { augmentationEnabled, integrationEnabled, isPristine, showAdvanced } = this.state;
+
+    return (
+      <div className={ styles.options }>
+        <p>
+          Integrate the Parity Ethereum client with the Chrome browser
+        </p>
+
+        <div className={ styles.option }>
+          <div className={ styles.switch }>
+            <Switch
+              checked={ integrationEnabled }
+              className={ styles.check }
+              onChange={ this.handleToggleIntegration }
+            />
+          </div>
+          <div>
+            Web3 Integration
+          </div>
+        </div>
+
+        <div className={ styles.option }>
+          <div className={ styles.switch }>
+            <Switch
+              checked={ augmentationEnabled }
+              className={ styles.check }
+              onChange={ this.handleToggleAugmentation }
+            />
+          </div>
+          <div>
+            Identity Augmentation
+          </div>
+        </div>
+
+        <br />
+
+        <div className={ styles.option }>
+          <div className={ styles.checkbox }>
+            <CheckBox
+              checked={ showAdvanced }
+              className={ styles.check }
+              onChange={ this.handleToggleAdvanced }
+            />
+          </div>
+          <div>
+            Show Advanced Options
+          </div>
+        </div>
+
+        { showAdvanced ? this.renderAdvancedOptions() : null }
+
+        <div className={ styles.buttons }>
+          <Button
+            disabled={ isPristine }
+            onClick={ this.handleSave }
+            primary
+          >
+            SAVE
+          </Button>
+
+          { showAdvanced ? this.renderAdvancedButtons() : null }
+        </div>
+      </div>
+    );
+  }
+
+  renderAdvancedButtons () {
+    return (
+      <div className={ styles.buttons }>
+        <Button onClick={ this.handleReset }>
+            DEFAULT
+          </Button>
+        <Button accent onClick={ this.handleClearCache }>
+          CLEAR CACHE
+        </Button>
+      </div>
+    );
+  }
+
+  renderAdvancedOptions () {
+    const { DAPPS, lookupURL, UI } = this.state;
+
+    return (
+      <div className={ styles.advanced }>
+        <div className={ [ styles.option, styles.optionInput ].join(' ') }>
+          <div className={ styles.input }>
+            <TextField
```

Comment:
I think it would be cool to perform a check if the service is accessible before saving the options.
If we cannot connect (or the output is invalid) we should display a warning (but not prevent from saving such options).